ydb/core/kqp/executer_actor/kqp_scheme_executer.cpp

```diff
@@ -92,6 +98,47 @@ class TKqpSchemeExecuter : public TActorBootstrapped<TKqpSchemeExecuter> {
         auto ev = MakeHolder<TEvTxUserProxy::TEvProposeTransaction>();
         auto& record = ev->Record;
 
+        record.SetDatabaseName(Database);
+        record.SetUserToken(NACLib::TSystemUsers::Tmp().SerializeAsString());
+        record.SetPeerName(ClientAddress);
+
+        auto* modifyScheme = record.MutableTransaction()->MutableModifyScheme();
+        modifyScheme->SetWorkingDir(GetTmpDirPath(Database));
+        modifyScheme->SetOperationType(NKikimrSchemeOp::EOperationType::ESchemeOpMkDir);
+        modifyScheme->SetAllowCreateInTempDir(false);
+        modifyScheme->SetInternal(true);
+
+        auto* makeDir = modifyScheme->MutableMkDir();
+        makeDir->SetName(GetSessionDirName());
+
+        auto* modifyAcl = modifyScheme->MutableModifyACL();
+        modifyAcl->SetName(GetSessionDirName());
+        modifyAcl->SetNewOwner(NACLib::TSystemUsers::Tmp().GetUserSID());
```

Comment:
`Name` and `NewOwner` are ignored if `ModifyACL` is a supplement to a main operation.

Setting `Name` just creates a noise in the code.
Setting `NewOwner` would not work at all. And there is no need for it -- object will belong the user who created it anyway.